djautotask/api_rest.py

```diff
@@ -534,9 +541,16 @@ def get_child_url(self, parent_id):
     def update(self, instance, changed_fields, parent_id=None):
         # Only for updating records with models in the DB, not Dummy syncs
         endpoint_url = self.get_child_url(parent_id)
-        body = self._format_request_body(instance, changed_fields)
+        body = self._format_fields(instance, changed_fields)
         return self.request('patch', endpoint_url, body)
 
+    def create(self, instance, **kwargs):
+        parent = kwargs.pop('parent')
+        endpoint_url = self.get_child_url(parent.id)
+        body = self._format_fields(instance, kwargs)
+        response = self.request('post', endpoint_url, body)
+        return response.get('itemId'), body
```

Comment:
It took me a while to figure out why you are passing the body back as the response, as that is very bad. We could get some very major errors and bugs that way. 

When I said you should pass the response into `update_or_create_instance`, I meant pass the response we get from Autotask. I thought both the [service call tickets child](https://webservices2.autotask.net/atservicesrest/swagger/ui/index#/ServiceCallTicketsChild/ServiceCallTicketsChild_CreateEntity) and the [service call tickets]() endpoints return the created record in the response, but it looks like only the main [ServiceCallTickets](https://webservices2.autotask.net/atservicesrest/swagger/ui/index#/ServiceCallTickets/ServiceCallTickets_Query) enpoint has that in the response, because its a query not a create operation. That's really annoying on their part.

If this worked like connectwise, a successfuly POST operation returns the record we just created, which would make our life very easy. Why does AT have to be dumb...

I think the safest way to handle this is to undo the changes you made from this comment: https://github.com/KerkhoffTechnologies/django-autotask/pull/151#discussion_r706500844 , while it isn't the most efficient request wise, it is the safest way to ensure we keep synchronized.

Reply:
Yeap, I thought you already had checked that response and asked that. Applied and review again.


cf. One more thing for our KB: 

> but it looks like only the main ServiceCallTickets enpoint has that in the response, because its a query not a create operation.

This post doc isn't relevant for ServiceCallTickets creation. See [this](https://webservices2.autotask.net/atservicesrest/swagger/ui/index#/ServiceCallTicketsChild/ServiceCallTicketsChild_CreateEntity) or [that one](https://git.ops.topleft.team/apps/django-topleft/-/issues/2131#note_32352), too.